src/core/receiver/control_thread.cc

```diff
@@ -92,23 +92,37 @@ ControlThread::ControlThread()
     const std::string empty_implementation;
     std::string src_impl = configuration_->property("SignalSource.implementation", empty_implementation);
     int src_count = configuration_->property("Receiver.sources_count", 1);
-    if (src_impl.empty() && (src_count != 1))
+    if (src_impl.empty())
         {
-            int num_src = 0;
-            for (int i = 0; i < src_count; i++)
+            if ((src_count == 1))
```

Comment:
this test is not necessary, and will cause config files with a single "SignalSource0.implementation" section to fail